daq/configurator.py

```diff
@@ -53,70 +53,69 @@ def print_config(config):
     print(*config_list, sep='\n')
 
 
-def merge_config(base, adding):
-    """Update a dict object and follow nested objects"""
-    if not adding:
-        return
-    for key in sorted(adding.keys()):
-        value = adding[key]
-        if isinstance(value, dict) and key in base:
-            merge_config(base[key], value)
-        else:
-            base[key] = copy.deepcopy(value)
-
-
-def load_config(path, filename=None, optional=False):
-    """Load a config file"""
-    if not path:
-        return None
-    config_file = os.path.join(path, filename) if filename else path
-    if not os.path.exists(config_file):
-        if optional:
-            LOGGER.info('Skipping missing %s', config_file)
-            return {}
-        raise Exception('Config file %s not found.' % config_file)
-
-    LOGGER.info('Loading config from %s', config_file)
-    with open(config_file) as data_file:
-        return yaml.safe_load(data_file)
-
-
-def load_and_merge(base, path, filename=None, optional=False):
-    """Load a config file and merge with an existing base"""
-    merge_config(base, load_config(path, filename, optional))
-
-
-def write_config(path, filename, config):
-    """Write a config file"""
-    if not path:
-        return
-    if not os.path.exists(path):
-        os.makedirs(path)
-    config_file = os.path.join(path, filename)
-    LOGGER.info('Writing config to %s', config_file)
-    with open(config_file, 'w') as output_stream:
-        output_stream.write(json.dumps(config, indent=2, sort_keys=True))
-        output_stream.write('\n')
-
-
 class Configurator:
     """Manager class for system configuration."""
```

Comment:
I think it should be cleaner to move config into configuration as well; then we can also move config retrieval methods here.